data/src/main/java/com/yapp/bol/data/datasource/file/FileDataSourceImpl.kt

```diff
@@ -0,0 +1,39 @@
+package com.yapp.bol.data.datasource.file
+
+import com.yapp.bol.data.model.group.response.ImageFileUploadResponse
+import com.yapp.bol.data.remote.FileApi
+import com.yapp.bol.data.utils.Image
+import com.yapp.bol.domain.handle.BaseRepository
+import com.yapp.bol.domain.model.ApiResult
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import okhttp3.MediaType
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import java.io.File
+import java.net.URLConnection
+import javax.inject.Inject
+
+class FileDataSourceImpl @Inject constructor(
+    private val fileApi: FileApi,
+) : BaseRepository(), FileDataSource {
+
+    override fun postFileUpload(file: File): Flow<ApiResult<ImageFileUploadResponse>> = flow {
+        val fileBody = RequestBody.create(MediaType.parse(getMimeType(file.name)), file)
+        val filePart = MultipartBody.Part.createFormData(FILE_KEY, file.name, fileBody)
+        val purpose = RequestBody.create(MediaType.parse(getMimeType(file.name)), Image.GROUP_IMAGE)
+
+        val result = safeApiCall {
+            fileApi.postFileUpload(file = filePart, purpose = purpose)
+        }
+        emit(result)
+    }
+
+    private fun getMimeType(fileName: String): String {
+        return URLConnection.guessContentTypeFromName(fileName)
+    }
```

Comment:
```suggestion
override fun postFileUpload(file: File): Flow<ApiResult<ImageFileUploadResponse>> = flow {
        val fileBody = file.asRequestBody(file.name.toMediaTypeOrNull())
        val filePart = MultipartBody.Part.createFormData(FILE_KEY, file.name, fileBody)
        val purpose = Image.GROUP_IMAGE.toRequestBody()

        val result = safeApiCall {
            fileApi.postFileUpload(file = filePart, purpose = purpose)
        }
        emit(result)
    }
```

build.gradle 에  `implementation(com.yapp.bol.Retrofit.LOGGING_INTERCEPTOR)` 을 추가하면 요렇게 개선해 볼 수 있을 것 같네요.

Reply:
사실 요건 제 작업사항이 아니라 재홍님과 말씀해보시죠 @prk4224 🙇‍♀️ 근데 개인적으로 궁금한건 요기 수정해주신 코드에서 logging interceptor가 하는 역할이 뭔가요?!

Reply:
된다면 괜찮네요 ! 직접 수정해주셔도 좋고 구조 리펙터링 PR 끝나면 추후에 적용해 볼게요 ~